pkg/lib/v0_2_0/keptn.go

```diff
@@ -71,12 +73,17 @@ func NewKeptn(incomingEvent *cloudevents.Event, opts keptn.KeptnOpts) (*Keptn, e
 	}
 	k.Logger = keptn.NewLogger(k.KeptnContext, incomingEvent.Context.GetID(), loggingServiceName)
 
+	if opts.Context == nil {
+		k.Context = context.Background()
+	}
+
```

Comment:
This code is not clear to me. I would expect something like the following:
```suggestion
    k.Context = context.Background()
	if opts.Context != nil {
		k.Context = opts.Context
	}

```

Reply:
Good catch, that was a bug. The intention is right in your code, the context should always have a default value and the user can provide their own via the options. Fixed it and added a test to verify it works as expected.